src/main/java/com/gooddata/md/maintenance/MaintenanceService.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2007-2016, GoodData(R) Corporation. All rights reserved.
+ */
+
+package com.gooddata.md.maintenance;
+
+import static com.gooddata.util.Validate.notNull;
+
+import com.gooddata.AbstractPollHandler;
+import com.gooddata.AbstractService;
+import com.gooddata.FutureResult;
+import com.gooddata.GoodDataException;
+import com.gooddata.GoodDataRestException;
+import com.gooddata.PollResult;
+import com.gooddata.gdc.TaskStatus;
+import com.gooddata.gdc.UriResponse;
+import com.gooddata.project.Project;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * Service providing metadata maintenance tasks.
+ */
+public class MaintenanceService extends AbstractService {
```

Comment:
I changed it back to Maintenance since we agreed on moving everything to package `maintenance` and introducing `MaintenanceService` so I thought the exception name could remain aligned.